pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2514,22 +2514,26 @@ private void handleDynamicConfigurationUpdates() {
                                 return;
                             }
                             Field configField = configFieldWrapper.field;
-                            Object newValue = FieldParser.value(data.get(configKey), configField);
-                            if (configField != null) {
-                                Consumer listener = configRegisteredListeners.get(configKey);
-                                try {
-                                    Object existingValue = configField.get(pulsar.getConfiguration());
+                            Consumer listener = configRegisteredListeners.get(configKey);
+                            Object newValue = null;
+                            Object existingValue;
+
+                            try {
+                                if (configField != null) {
```

Comment:
```suggestion
                            Object existingValue;

                            try {
                                final Object newValue;
                                if (configField != null) {
```

and you don't need to print `newValue` in the catch block because it might be null